### DISCUSSION TITLE
[Formula author] Benefit of instructing users to run `tap` first?

### DISCUSSION THREAD

**Discussion by jpmcb** [Homebrew/discussions#2243] [Writing Formulae/Casks] [answered]
I've noticed that almost all projects instruct users to do the following:
```
$ brew tap github-org/repo-name
$ brew install my-formula
```
but I've also noticed that you can do:
```
$ brew install github-org/repo-name/my-formula
```
which appears to do a tap first and then installs the repo.

What's the benefit of instructing users to run a tap first? I couldn't find much documentation around this or relevant issues / discussions here. Happy to be pointed to some docs! Thanks all

**Comment by carlocab** [comment, marked as answer]
If your tap is on GitHub, then there aren't really any benefits to doing `brew tap` before `brew install`, other than not having to pass the fully-qualified `github-org/tap-name/formula-name` to `brew install`. (That benefit is, admittedly, dubious.)

If your tap is not hosted on GitHub (e.g. GitLab), then you need to do `brew tap` first, because doing `brew install` will have `brew` looking up either a wrong or non-existent tap.

**Reply by jpmcb** [reply]
Thanks much! That's helpful context!!
